### PR TITLE
Expand the capabilities of dictionaries

### PR DESCRIPTION
**Pull request by vkatsuba**
* Added config option `default_dictionary` for use any custom dictionary
* Added config option `additional_dictionaries` for expand exist dictionary

If set `default_dictionary` only - will be used `default_dictionary` instead of the dictionary that is currently in the source code.
If set `additional_dictionaries` only - will be used dictionary that is currently in the source code and added additional dictionaries to it as an extension.
If set `default_dictionary` and `additional_dictionaries` - will be used default dictionary what was set plus additional dictionaries to it as an extension.

### CODE REVIEW

File: src/sheldon_dictionary.erl [hunk 1]
```diff
@@ -168,3 +168,30 @@ typo_model(Word) ->
                 end,
                 [Word],
                 [lists:split(I, Word) || I <- lists:seq(0, length(Word))]).
+
+-spec generate_dictionary(term()) -> binary().
```
Comment (elbrujohalcon) [inline src/sheldon_dictionary.erl:172]:
```suggestion
-spec generate_dictionary(file:name_all()) -> binary().
```

File: src/sheldon_dictionary.erl [hunk 2]
```diff
@@ -168,3 +168,30 @@ typo_model(Word) ->
...
+        {undefined, [_ | _] = AdditionalDictionaries} ->
+            concat_dictionaries(Source, AdditionalDictionaries);
+        {[_ | _] = DefaultDictionary, []} ->
+            {ok, SourceBin} = file:read_file(DefaultDictionary),
+            SourceBin;
+        {[_ | _] = DefaultDictionary, [_ | _] = AdditionalDictionaries} ->
+            concat_dictionaries(DefaultDictionary, AdditionalDictionaries)
+    end.
+
+-spec concat_dictionaries(list(), list() | []) -> binary().
```
Comment (elbrujohalcon) [inline src/sheldon_dictionary.erl:189]:
```suggestion
-spec concat_dictionaries(file:name_all(), list() | []) -> binary().
```

File: src/sheldon_dictionary.erl [hunk 3]
```diff
@@ -168,3 +168,30 @@ typo_model(Word) ->
...
+generate_dictionary(Source) ->
+    DefaultDictionary = application:get_env(sheldon, default_dictionary, undefined),
+    AdditionalDictionaries = application:get_env(sheldon, additional_dictionaries, []),
+    case {DefaultDictionary, AdditionalDictionaries} of
+        {undefined, []} ->
+            {ok, SourceBin} = file:read_file(Source),
+            SourceBin;
+        {undefined, [_ | _] = AdditionalDictionaries} ->
+            concat_dictionaries(Source, AdditionalDictionaries);
+        {[_ | _] = DefaultDictionary, []} ->
```
Comment (elbrujohalcon) [inline src/sheldon_dictionary.erl:182]:
```suggestion
        {DefaultDictionary, []} ->
```
Filenames can be binaries, among other things… It's not necessary to use the robot butt here nor in the next clause.

File: src/sheldon_dictionary.erl [hunk 4]
```diff
@@ -168,3 +168,30 @@ typo_model(Word) ->
                 end,
                 [Word],
                 [lists:split(I, Word) || I <- lists:seq(0, length(Word))]).
+
+-spec generate_dictionary(term()) -> binary().
+generate_dictionary(Source) ->
+    DefaultDictionary = application:get_env(sheldon, default_dictionary, undefined),
```
Comment (elbrujohalcon) [inline src/sheldon_dictionary.erl:174]:
```suggestion
    DefaultDictionary = application:get_env(sheldon, default_dictionary, Source),
```
I guess this change ☝🏻 would simplify the `case` statement below.

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]
A minimal spec change and some refactoring, but the PR in general looks good.

File: src/sheldon_dictionary.erl (hunk 2 above)
Comment (vkatsuba) [reply src/sheldon_dictionary.erl]: Done

File: src/sheldon_dictionary.erl (hunk 4 above)
Comment (vkatsuba) [reply src/sheldon_dictionary.erl]: Nice! Done.

**Review by elbrujohalcon — APPROVED** [review]
Excellent!

**Pull request merged**